### ISSUE TITLE
Clean install is weirdly borked

### ISSUE THREAD

**Issue by JamesYeoman** [eirslett/storybook-builder-vite#232]
So I used `npx sb@next repro`, and it all works fine. But as I've commented in the following code block, a clean vite + sb-builder-vite install errors out

```bash
yarn create vite # I used the react-ts template
cd <folder name>
yarn
npx sb@next init --builder storybook-builder-vite # https://storybook.js.org/blog/storybook-for-vite/
yarn storybook # acorn jsx error
yarn add --dev @storybook/core-common webpack@^5 # Peer dependencies that yarn said were unmet, so this is an attempt to try to fix the error
yarn storybook # acorn jsx error still occuring
```

Here's a screenshot of the "acorn jsx error"
![image](https://user-images.githubusercontent.com/21042602/153189653-c7216040-6300-4a62-976e-e1211b5382d1.png)

This error seems to make everything load infinitely. No error gets displayed in my shell, and everything comes to a standstill.

**Comment by joshwooding** [comment]
I think this happens when two incompatible versions of acorn are used. I'll see if I can reproduce

**Comment by JamesYeoman** [comment]
Here's a repo with the reproduction up to the first `yarn storybook` https://github.com/JamesYeoman/sb-vite-bug-repro if it helps. I'm also using node v16.10.0 and yarn v1.22.17, all installed using [volta](https://volta.sh/) on Windows 10

**Comment by wayne-o** [comment]
I am getting this on a clean install - bit of a blocker :/

**Comment by IanVS** [comment]
OK, thanks for the reproduction @JamesYeoman.  I've figured out what's happening.  If `acorn-jsx` is not hoisted up to the top-level node_modules directory, it isn't being pre-bundled (see https://github.com/vitejs/vite/issues/3024), and for whatever reason (probably changes to the dependencies used inside storybook), `acorn-jsx` is now only inside `node_modules/@storybook/addon-docs/node_modules/acorn-jsx`, so it isn't being processed/pre-bundled.  

The temporary solution is to add `'@storybook/addon-docs > acorn-jsx'` to your `optimizeDeps.include` in `viteFinal` (being careful not to overwrite the existing values in the config). I'm working on adding it to our base config, but this is tricky to do with this ` > ` syntax.  I think I've found a way, though, and should have a PR up shortly.
